src/peer_bootstrap_node.cpp

```diff
@@ -25,40 +50,75 @@ void PeerBootstrapNode::sendData(const std::vector<uint8_t> &data)
     // Data sending code here
 }
 
-void PeerBootstrapNode::receiveData(std::vector<uint8_t> &data)
+void PeerBootstrapNode::receiveData(const asio::error_code &error, std::size_t bytes_transferred)
 {
-    Logger::log(LogLevel::DEBUG, "Receiving data...");
-    // Data receiving code here
-}
+    if (!error && bytes_transferred > 0)
+    {
+        std::vector<uint8_t> data(receive_buffer_.begin(), receive_buffer_.begin() + bytes_transferred);
+        Logger::log(LogLevel::INFO, "Received data from " + remote_endpoint_.address().to_string() + ":" +
+                                        std::to_string(remote_endpoint_.port()) +
+                                        " Size: " + std::to_string(bytes_transferred));
+
+        // Process the data (placeholder)
+        Logger::log(LogLevel::DEBUG, "Data: " + std::string(data.begin(), data.end()));
 
+        // Continue listening
+        socket_.async_receive_from(
+            asio::buffer(receive_buffer_), remote_endpoint_,
+            [this](const asio::error_code &error, std::size_t bytes_transferred)
```

Comment:
Why use a lambda around the function instead of the actual function

Reply:
it's kinda of a standard for this part.